src/r-exasol/impl/rconnection/RReaderConnection.cpp

```diff
@@ -0,0 +1,74 @@
+//
+// Created by thomas on 08/07/2021.
+//
+
+#include <r-exasol/impl/rconnection/RReaderConnection.h>
+#include <r-exasol/if/Reader.h>
+#include <cstdlib>
+
+
+extern "C" {
+
+#define class clss //need to hide class as it is reserved in C++, but used as variable name in Connections.h
+#define private priv //need to hide private as it is reserved in C++, but used as variable name in Connections.h
+#include <R_ext/Connections.h>
+#undef class
+#undef private
+
+extern int dummy_fgetc(Rconnection con);
+extern void Rf_set_iconv(Rconnection con);
+
+
+static size_t pipe_read(void *ptr, const size_t size, const size_t nitems,
+                        const Rconnection con) {
+    size_t retVal = 0;
+    exa::reader::Reader * reader = *((exa::reader::Reader **) con->priv);
+    if(reader) {
+        retVal = reader->pipe_read(ptr, size, nitems);
+    }
+    return retVal;
+}
+
+static int file_fgetc_internal(const Rconnection con) {
+    exa::reader::Reader * reader = *((exa::reader::Reader **) con->priv);
+    int retVal = 0;
+    if(reader) {
+        retVal = reader->fgetc();
+    }
+    return  retVal;
+}
+
+}
+
+namespace rcon = exa::rconnection;
+
+rcon::RReaderConnection::RReaderConnection(reader::Reader & reader)
+: mReader(reader)
+, mConn(nullptr){}
+
+SEXP rcon::RReaderConnection::create() {
+    SEXP r_custom_connection;
+    PROTECT(r_custom_connection = R_new_custom_connection("exasol", "r", "textConnection", &mConn));
```

Comment:
I guess this is the pipe which we use to connect the R world with the http socket. We should document this somewhere.

Reply:
I added some documentation to the code now